### ISSUE TITLE
GenericLikelihoodModel bootstrap requires exog

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#944]
calling bootstrap with a GenericLikelihood Results instance that doesn't have an exog raises an exception.

**Comment by mtzl** [comment]
Is this still valid? #1434 has been closed, but when using an MLE without an exog (I'm fitting a mixture model), an exception is still raised, because [this line](https://github.com/statsmodels/statsmodels/blob/master/statsmodels/base/model.py#L2223) attempts to slice `None`.

Also, this line has some other minor issues:
- the model instance is not cloned correctly: `hascloneattr` inspects the result for cloneable attributes, not the model (I have some extra arguments for the init to set attributes of my MLE subclass -- instances of scipy.stats.rv_continuous distributions to be used in the LLH) 
- having `exog=None` breaks `result.summary()`, since some values (`result.df_resid`) are NaN, and the  string representation (`lambda: ["%#6d" % results.nobs])`) can't deal with NaN.

I have [a pull request](https://github.com/statsmodels/statsmodels/pull/4676) for these

**Comment by josef-pkt** [comment]
I don't see any change in 1434 that would affect this. (maybe I never committed that part)
So this issue still open.

nobs and df_resid: besides making summary more robust to nan and float, df_resid and nobs should be properly defined also in this case
df_resid is also still required for t_test (even when it's not used)

**Comment by josef-pkt** [comment]
BUG for df_resid

GenericLikelihoodModelResults checks for existence of df_model, but the model `initialize` sets df_resid to nan if exog is None.

```
        if hasattr(model, 'df_model'):
            self.df_model = model.df_model
        else:
            self.df_model = len(mlefit.params)
```

something strange: 
I remember the setting of df_resid in GenericLikelihoodModelResults to be more complicated than it is in the current master, e.g. checking for the length of params and adjusting the exog_names parameter names for use in summary `xname`.

**Comment by josef-pkt** [comment]
I'm closing this
bootstrap with `exog=None` works in #7488 (after I fix init_keys to recreate model, no exog related problems)

res,summary() also works

df_resid, df_model look fine but I haven't investigated much
(only aic based on df_modelwc was wrong without k_constant, but I fix that also in #7488)
